File: LeetCode/JavaScript/#485 Max Consecutive Ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
	var res = 0;
	var temp;
    nums = nums.join("");
    while(nums.indexOf(0)!==-1){
    	temp = nums.search(0);
    	if(res<temp){
    		res = temp;
    	}
    	nums = nums.slice(temp+1);
    }
    return Math.max(res,nums.length);
};

File: LeetCode/Python/#485 Max Consecutive Ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self,nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        this_val = 0
        max_val = 0

        for i in nums:
            if i==1:
                this_val+=1
            else:
                this_val = 0
            max_val = max(this_val,max_val)
        return max_val
